src/component/DisplayRecipe.js

```diff
@@ -1,19 +1,21 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useContext } from "react";
 import RecipeDetails from "./RecipeDetails";
 import RecipeIngredients from "./RecipeIngredients";
 import RecipeDirection from "./RecipeDirection";
 import * as api from "../api";
+import { NavigationContext } from "../context/NavigationContext";
 
-function DisplayRecipe({ recipeId }) {
+function DisplayRecipe() {
   const [selectedRecipe, setSelectedRecipe] = useState(null);
 
   const [selectedServings, setSelectedServings] = useState(
     selectedRecipe?.servings || 0
   );
+  const { currentPath } = useContext(NavigationContext);
+  const recipeId = currentPath.substring(9);
```

Comment:
There is a general practice that there shouldn't be any magic number in the code. Please assign number `9` with a constant, which should say why 9 is being used.